mesos-rxjava-recordio/src/test/java/org/apache/mesos/rx/java/recordio/RecordIOOperatorTest.java

```diff
@@ -72,186 +56,115 @@ public void correctlyAbleToReadEventsFromEventsBinFile() throws Exception {
-        });
+        final List<Event> events = runTestOnChunks(bufChunks);
+        assertThat(events).isEqualTo(EVENT_PROTOS);
     }
 
     @Test
     public void readEvents_multipleEventsInOneChunk() throws Exception {
-        runTestOnChunks(newArrayList(
-            Unpooled.copiedBuffer(ByteArrays.concatAllChunks(subHbOffer))
-        ), (subscriber, recordIOSubscriber, events) -> {
-            assertThat(events).hasSize(3);
-            assertThat(events.get(0).getType()).isEqualTo(Event.Type.SUBSCRIBED);
-            assertThat(events.get(1).getType()).isEqualTo(Event.Type.HEARTBEAT);
-            assertThat(events.get(2).getType()).isEqualTo(Event.Type.OFFERS);
-        });
-    }
-
-    @Test
-    public void readEvents_singleEvent_chunkSize_60() throws Exception {
-        final int numPartitions = 1;
-        final byte[] chunk = RecordIOUtils.eventToChunk(TestingProtos.SUBSCRIBED);
-        runTestOnPartitionedChunk(chunk, numPartitions);
-    }
-
-    @Test
-    public void readEvents_singleEvent_chunkSize_30() throws Exception {
-        final int numPartitions = 2;
-        final byte[] chunk = RecordIOUtils.eventToChunk(TestingProtos.SUBSCRIBED);
-        runTestOnPartitionedChunk(chunk, numPartitions);
-    }
-
-    @Test
-    public void readEvents_singleEvent_chunkSize_20() throws Exception {
-        final int numPartitions = 3;
-        final byte[] chunk = RecordIOUtils.eventToChunk(TestingProtos.SUBSCRIBED);
-        runTestOnPartitionedChunk(chunk, numPartitions);
-    }
-
-    @Test
-    public void readEvents_singleEvent_chunkSize_12() throws Exception {
-        final int numPartitions = 5;
-        final byte[] chunk = RecordIOUtils.eventToChunk(TestingProtos.SUBSCRIBED);
-        runTestOnPartitionedChunk(chunk, numPartitions);
-    }
-
-    @Test
-    public void readEvents_singleEvent_chunkSize_06() throws Exception {
-        final int numPartitions = 10;
-        final byte[] chunk = RecordIOUtils.eventToChunk(TestingProtos.SUBSCRIBED);
-        runTestOnPartitionedChunk(chunk, numPartitions);
+        final List<Event> subHbOffer = newArrayList(
+            TestingProtos.SUBSCRIBED,
+            TestingProtos.HEARTBEAT,
+            TestingProtos.OFFER
+        );
+        final List<byte[]> eventChunks = RecordIOUtils.listMap(subHbOffer, RecordIOUtils::eventToChunk);
+        final List<ByteBuf> singleChunk = newArrayList(Unpooled.copiedBuffer(ByteArrays.concatAllChunks(eventChunks)));
+
+        final List<Event> events = runTestOnChunks(singleChunk);
+        assertThat(events).isEqualTo(subHbOffer);
     }
 
     @Test
-    public void readEvents_singleEvent_chunkSize_03() throws Exception {
-        final int numPartitions = 15;
+    public void readEvents_singleEvent_variousChunkSizes() throws Exception {
+        final List<Integer> partSizes = newArrayList(1, 2, 3, 4, 5, 10, 20, 30, 40, 50, 60);
```

Comment:
I want to make sure that we keep getting results for every scenario.

If one of partition sizes were to fail none of the others would run.  In order to keep the results we could have one method for each value like before, or could use JUnits [Parameterized tests](https://github.com/junit-team/junit/wiki/Parameterized-tests) feature.